### PR TITLE
http: implement FailIfRegexp and FailIfNotRegexp

### PR DESCRIPTION
**Pull request by stapelberg**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -29,11 +30,13 @@ type Module struct {
...
-	NoFollowRedirects bool   `yaml:"no_follow_redirects"`
-	FailIfSSL         bool   `yaml:"fail_if_ssl"`
-	FailIfNotSSL      bool   `yaml:"fail_if_not_ssl"`
-	Method            string `yaml:"method"`
+	ValidStatusCodes  []int    `yaml:"valid_status_codes"`
+	NoFollowRedirects bool     `yaml:"no_follow_redirects"`
+	FailIfSSL         bool     `yaml:"fail_if_ssl"`
+	FailIfNotSSL      bool     `yaml:"fail_if_not_ssl"`
+	Method            string   `yaml:"method"`
+	FailIfRegexp      []string `yaml:"fail_if_regexp"`
```
Comment (brian-brazil) [inline main.go]: Might `fail_if_matches_regexp` be clearer?

Comment (stapelberg) [inline main.go]: As it contains more information, it is indeed easier to understand. I was shooting for consistency with the already-existing FailIfSSL. Do you want me to change it to FailIfMatchesRegexp?

Comment (brian-brazil) [inline main.go]: Sounds good.

**Comment by stapelberg** [comment]
Sorry for the late reply, I’ve been busy.

I’ve changed the name of the option as you asked for. PTAL.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**